sycl/include/CL/sycl/detail/defines.hpp

```diff
@@ -31,3 +37,12 @@
 #ifndef SYCL_EXTERNAL
 #define SYCL_EXTERNAL
 #endif
+
+#if defined(__SYCL_ID_QUERIES_FIT_IN_INT__) && __has_builtin(__builtin_assume)
+#define __SYCL_ASSUME_INT(x) __builtin_assume(x <= INT_MAX)
```

Comment:
```suggestion
#define __SYCL_ASSUME_INT(x) __builtin_assume((x) <= INT_MAX)
```

Reply:
if x is some expression it is safer to have it as (x) here.

Reply:
Nice catch. Didn't notice this thing here.